### ISSUE TITLE
"Add row to table" confusing design pattern

### ISSUE THREAD

**Issue by moshee** [LibrePCB/LibrePCB#1424]
##### SUMMARY

This design pattern is used in multiple places on the GUI, where you have a table with an add button in the last column:

![image](https://github.com/user-attachments/assets/409380d7-8d3f-4aec-8a03-926e7171e5f6)

1. In the pictured scenario, the next button is disabled because the table actually has zero rows currently, even though I have already made the necessary selection to proceed (chosen a symbol). This is very unclear in context.
    * The "New" text is there, but it's not immediately clear what this means with zero rows in the table. I keep thinking it means that this is a new row that has already been added to the table, or something.
2. The add row button is clunky as a table cell. When you click it, it moves out from under the cursor, so you cannot click multiple times in a row as easily.
3. It is misleading to have a row that contains data but is not actually in the table. If I can see the row, it should already be in the table.

##### SOLUTION

1. If a table requires at least one row to be valid, then the row should be already present. If cells must not be empty, validators can format the cells accordingly.
2. Add row button should ideally be placed out of band/in some static location not mixed with the cell data.
3. Get rid of the special "New:" row. If in-band signaling for a possible new row is desired, the formatting should be very different to make it obvious that it is a UI cue and not a data row. For example, a row with greyed out italic text saying "*Click to add new row*" and no other cells present.

**Comment by ubruhin** [comment]
I agree - the main reason for the current design is that with the QtWidgets framework it's very hard to build something better. Even if there are better ideas available, the implementation with QtWidgets can be very hard or almost impossible. It is one reason why I am currently evaluating another, more flexible GUI toolkit.

**Comment by ubruhin** [comment]
The new UI (see #1494) replaces this design pattern with a more intuitive concept. There are sill a few old tables there, but they will be replaced step by step. So I think we can close this.
